core/trino-main/src/main/java/io/trino/execution/scheduler/StageTaskSourceFactory.java

```diff
@@ -353,8 +363,13 @@ public void close()
         private final Optional<CatalogName> catalogRequirement;
         private final long targetPartitionSourceSizeInBytes; // compared data read from ExchangeSources
         private final long targetPartitionSplitWeight; // compared against splits from SplitSources
+        private final Executor executor;
 
+        @GuardedBy("this")
+        private ListenableFuture<List<LoadedSplits>> loadedSplitsFuture = immediateFuture(ImmutableList.of());
```

Comment:
set it to `null` by default and check whether it is `null` when `getMoreTasks` is called